doc/changelog/04-tactics/17832-master+rewstrategy-explicit-assoc.rst

```diff
@@ -0,0 +1,8 @@
+- **Changed:**
+  In :tacn:`rewrite_strat`, the :g:`choice` strategy has a new
+  precedence: it now takes as arguments atomic rewriting
+  strategies, so a different parenthesizing may be required. No
+  incompatibilities on our continuous integration tests have been
+  detected though
```

Comment:
```suggestion
  In :tacn:`rewrite_strat`, the syntax for the :g:`choice` strategy has
  changed slightly.  You may need to add parentheses around its arguments.
  We didn't find any cases in which this change broke the continuous
  integration tests
```

How abort starting with the benefits of the PR?  You've only stated the costs here.
You're sure its OK to be a little incompatible without a deprecation step?

Reply:
Deprecation is important when there's a large cost to fix something, and especially important when there's no backwards compatible way of fixing the changes.  (In cases where the deprecation has a backwards compatible replacement, it's still useful to have for at least one version in cases where there may be many changes that could be automated if a deprecation warning were emitted.)  In this case, any incompatibility is easily fixed in a way that is both backwards compatible and improves code clarity, and furthermore I expect the number of incompatibilities to be small, so I think it's fine to skip deprecation.

The benefit is that the grammar now forces you to use non-confusing syntax for `choice` arguments in cases where the previous grammar allowed confusing and hard-to-disambiguate syntaxes.